src/pkg/mcp/tests/client_test.go

```diff
@@ -568,10 +543,8 @@ func TestInProcessMCPServer(t *testing.T) {
 	}
 
 	// Suite-level setup
-	var cleanup func()
-	projectDir, cleanup = createTestProjectDir(t)
+	projectDir = createTestProjectDir(t)
```

Comment:
We pass `t` to the function, so there's no reason it couldn't call its own `t.Cleanup`